plugins/callback/mail.py

```diff
@@ -94,34 +95,36 @@ def set_options(self, task_keys=None, var_options=None, direct=None):
         self.cc = self.get_option('cc')
         self.bcc = self.get_option('bcc')
 
+    @staticmethod
+    def encode_addresses(addresses):
```

Comment:
Why is this function necessary at all?

Reply:
It is needed to properly encode names that contain non-ASCII characters.

For ex. if I put
```ini
[callback_mail]
to = Lénaïc Huard <my@email.address.com>
```
in my `ansible.cfg`,

without this encoding function, this address would look like this in my mail user agent:
```
LÃ©naÃ¯c Huard <my@email.address.com>
```

Reply:
I'm not sure why you are not using `email.utils.getaddresses([addresses])` instead of `email.utils.parseaddr(x) for x in addresses_list`. That handles parsing of more complex addresses strings like `"a, b" <foo@bar.org>, "c, d" <bla@xyz.com>` correctly, as opposed to your current function:
```
>>> email.utils.getaddresses(['"a, b" <foo@bar.de>, "c, d" <bla@xyz.com>'])
[('a, b', 'foo@bar.de'), ('c, d', 'bla@xyz.com')]

>>> [email.utils.parseaddr(x) for x in '"a, b" <foo@bar.de>, "c, d" <bla@xyz.com>'.split(',')]
[('', 'a'), ('', 'b'), ('', 'c'), ('', 'd')]
```
(You would also have to get rid of `.split(',')` further below. Probably having another helper function for that would be good, that does something like `[email.utils.formataddr((''. address), charset='utf-8') for name, address in email.utils.getaddresses([addresses])]`.)

Anyway, both your change and my suggestion have some drawback: if the email address contains a non-ASCII character, it will fail. In times of IDNs, I'm not sure this is acceptable. It might be better if this goes into another PR, so that the main bugfix (Python 3 incompatibility) can get merged ASAP and this one can be discussed in more detail (it's also a bugfix, but less urgent).

Reply:
(https://en.wikipedia.org/wiki/International_email)

Reply:
I’ve just updated the PR to get rid of the `.split(',')` and leverage `email.utils.getaddresses`:
```diff
     @staticmethod
     def encode_addresses(addresses):
-        addresses_list = addresses.split(',')
-        addresses_list = [email.utils.formataddr(email.utils.parseaddr(x), charset='utf-8') for x in addresses_list]
+        addresses_list = [email.utils.formataddr((name, address), charset='utf-8') for name, address in email.utils.getaddresses([addresses])]
         return ','.join(addresses_list)
```

Let me know if that looks fine for you or if you still prefer to extract this change in a dedicated PR.

Reply:
A similar change is also required in lines 116, 118 and 120. (It's probably best to have the split required there as a separate function.)

Since this is a more complex change I think it's better to have it in a new PR.

Reply:
I’ve prepared #4061 to address this specific issue.